### ISSUE TITLE
Delete Resource needs to give more context to user

### ISSUE THREAD

**Issue by darctur** [hydroshare/hydroshare#2712]
The 'delete resource' tool provides a warning message that tells the user “This will delete any resources you have selected” but gives no other context for the user to know what will happen when he/she confirms the deletion. This can lead to accidental deletion when the user doesn't realize some resources are selected that were not intended to be deleted. (THIS HAS HAPPENED)

It might not make sense to enumerate every resource selected, but this seems serious enough that some information about the number of resources should be given in the warning dialog; perhaps list the first few, followed by “and 4 more resources selected” or whatever the number. The use case here is not about fully describing what’s about to be deleted, but to indicate the size of the selected set, with some identified, so the user has some context for a decision. Right now, there is NO context for a decision, just the red warning box, whether anything’s selected at all. 
 
One other thought is if there’s a way to indicate “this resource is not to be deleted” as a safeguard against accidents like this? ArcGIS Online lets me do that with map and feature services.

**Comment by aphelionz** [comment]
Additionally, we should consider a "soft delete" as @rayi113 has suggested. This would move resources to a category that hides it from public view on the site, but keeps the resources in a "Deleted" or "Trash" section of the admin, to be purged or preserved as seen fit, or on a timed basis.

**Comment by rayi113** [comment]
Credit to @mjstealey for suggesting the iRODS "soft delete" to me that I passed along.

**Comment by Castronova** [comment]
I agree that we should implement a "soft" delete, but recognize that we probably don't want to make data recovery a daily task.  We should also think about any modifications to the message below that might help prevent accidental deletes.  Below is the current message that's displayed when resources are deleted.  



![image](https://user-images.githubusercontent.com/4822372/38809728-855054c6-4152-11e8-8873-5f54a610bf50.png)

Suggested improvements:
- Give an extra warnings if a resource is marked for deletion and: 
    -  is `public` 
    - has multiple owners
    - has "enhanced" metadata, e.g. coauthors, extended metadata, funding agency, etc.  
- Add a `download copies` button in this dialog to encourage users to download a copy of the data they plan to delete.

**Comment by horsburgh** [comment]
Do we just need to implement a trash? Stuff in the trash would still count against quota (since we have to keep it on disk), but would be recoverable.

**Comment by mjstealey** [comment]
> Do we just need to implement a trash? Stuff in the trash would still count against quota (since we have to keep it on disk), but would be recoverable.

### 1.  Do we just need to implement a trash?

From an iRODS standpoint the concept of a **trash** already exists, and HydroShare presently avoids it's usage by including the `-f` or `-rf` flags when removing a resource via the `irm` call. 

- iRODS out of the box has a notion of `Zone/home/USERS` and `Zone/trash/home/USERS`
- When a resource is removed using `irm` the resource is moved from the location in `Zone/home/USERS` to the corresponding location in `Zone/trash/home/USERS` where it will remain until trash collection is performed via `irmtrash` or `irm -f` or `irm -rf` calls are made on the resource in the trash.
- HydroShare presently bypasses the use of a trash by removing resources with the `-f` or `-rf` flags. This was a conscious decision and made sense at the time, but review is now needed.

### 2. Stuff in the trash would still count against quota

This is a decision to be made by the team and could be implemented either way.

- If resources in the trash have an expiry date, then the quota issue is somewhat transient based on a rolling window of time.

## Implementing a "soft delete"

There are a few elements to keep in mind if this is to be implemented.

1. How to track **deleted but not yet purged** items in the Django database
2. How long to keep **deleted resources in the iRODS trash** prior to purging them
3. Define the **request to un-delete resource workflow** in an easy to maintain way

From iRODS, the trash maintenance can be accomplished via simple rules based on time spent in the trash or more brute force via a periodic script that reaches into the iRODS trash, calculates time of delete, and force remove/purge when resources have reached their expiry.

From Django's point of view I've personally lost track of the moving parts, but would think that some kind of flag could be introduced that tracks a deleted state and removes the resource from any user interaction while retaining the database entry until the resource expires from the trash and is fully purged.

As for the un-delete workflow, you may not want to make that a well known "feature" and only use it when the **OMG!!!... my data!** requests come in.

**Comment by aphelionz** [comment]
Agree with @Castronova 👍 for points that are all immediately actionable in the UI

**Comment by Castronova** [comment]
related to #2743
